layers/+completion/helm/funcs.el

```diff
@@ -55,465 +59,266 @@
+                                         (mapcar #'buffer-file-name
+                                                 (buffer-list)))))
+               ("dir" (seq-filter #'file-regular-p
+                                  (directory-files (if prefix-p
+                                                       (ask-dir-name)
+                                                     default-directory)
+                                                   t)))
+               ("project" (seq-filter #'file-regular-p
+                                      (directory-files (if prefix-p
+                                                           (ask-project-dir)
+                                                         (current-project-root))
+                                                       t))))))
+         (helm-grep-wrapper targets default-input-p)))
+      ((pred (lambda (x) (member x '("rg" "ag" "pt" "ack" "grep"))))
+       (require 'helm-ag)
+       (pcase-let (;; setup appropriate base command and exit status
+                   (helm-ag-base-command
+                    (pcase search-tool
+                      ("rg"
+                       (concat "rg --smart-case --no-heading --color=never --line-number"
+                               (when-let
+                                   ((maxcol spacemacs-helm-rg-max-column-number)
+                                    (_ (> maxcol 0)))
+                                 (format " --max-columns=%d" maxcol))))
+                      ((and "ag" (pred (lambda (_) (memq system-type '(ms-dos windows-nt)))))
+                       "ag --vimgrep")
+                      (val
+                       (format "%s --nocolor --nogroup" (if (string= val "pt")
+                                                            "pt -e"
+                                                          val)))))
+                   (helm-ag-success-exit-status
+                    (when (string= search-tool "rg")
+                      (list 0 2)))
+                   ;; function to call and additional args
+                   (`(,func ,args)
+                    (pcase scope
+                      ("buffer" (list (if prefix-p
+                                          #'helm-do-ag-this-file
+                                          #'helm-do-ag-buffers)
+                                      nil))
+                      ("dir" (list #'helm-do-ag
+                                   (if prefix-p
+                                       (ask-dir-name)
+                                     default-directory)))
+                      ("project" (list #'helm-do-ag
+                                       (if prefix-p
+                                           (ask-project-dir)
+                                         (current-project-root)))))))
+         (helm-ag-wrapper func default-input-p args)))
+      (_ (user-error "Not a valid search tool %s" search-tool)))))
+
+(byte-compile 'spacemacs//helm-search-dispatcher)
```

Comment:
This should be removed, otherwise you compile each time the file is loaded. Instead we should rely on the ability to auto-compile all spacemacs files, this is a setting in your dotfile.